common/network-common/src/main/java/org/apache/spark/network/buffer/FileSegmentManagedBuffer.java

```diff
@@ -77,16 +77,16 @@ public ByteBuffer nioByteBuffer() throws IOException {
         return channel.map(FileChannel.MapMode.READ_ONLY, offset, length);
       }
     } catch (IOException e) {
+      String errorMessage = "Error in reading " + this;
       try {
         if (channel != null) {
           long size = channel.size();
-          throw new IOException("Error in reading " + this + " (actual file length " + size + ")",
```

Comment:
This is just thrown and then ignored. I assigned it to `errorMessage` so that we can see it in the error.